pkg/app/piped/analysisprovider/metrics/prometheus/prometheus.go

```diff
@@ -73,6 +74,10 @@ func (p *Provider) RunQuery(ctx context.Context, query string, expected config.A
 	p.logger.Info("run query", zap.String("query", query))
 	// TODO: Use HTTP Basic Authentication with the username and password when needed.
 	response, warnings, err := p.api.Query(ctx, query, time.Now())
+	if errors.Is(err, context.DeadlineExceeded) {
+		// Treat as a success because query duration is already ended.
+		return true, nil
+	}
```

Comment:
IMO, this context checking should not be the responsibility of this function.
This will ignore the Deadline from both parent and child (set via line 71) contexts.

Reply:
Absolutely. It ends with failure when a request exceeds the timeout.